app/uk/gov/hmrc/customs/declaration/services/StatusResponseValidationService.scala

```diff
@@ -67,20 +70,37 @@ class StatusResponseValidationService @Inject() (declarationsLogger: Declaration
       declarationsLogger.errorWithoutRequestContext("Status response BadgeId field is missing")
       Left(ErrorResponse.errorBadRequest("Badge Identifier is missing or invalid"))
     })({ communicationAddress =>
-      if(badgeIdentifier.value.toUpperCase != extractBadgeIdentifier(communicationAddress.head).toUpperCase){
+      //      communicationAddress match {
+      //        case x if (x.isEmpty) => Left(ErrorResponse.errorBadRequest("Badge Identifier is missing or invalid"))
+      //        case x if (badgeIdentifier.value.toUpperCase != extractBadgeIdentifier(x.head).toUpperCase) => Left(ErrorResponse.errorBadRequest("Badge Identifier is missing or invalid"))
+      //
+      //        case seq: IndexedSeq[_] =>
+      //        case seq: LinearSeq[_] =>
+      //        case seq: NodeSeq =>
+      //        case _ =>
+      //      }
+      if (badgeIdentifier.value.toUpperCase != extractBadgeIdentifier(communicationAddress.headOption.getOrElse("")).toUpperCase) {
```

Comment:
This introduces a regression where we don't catch a missing badge identifier if `badgeIdentifier.value` happens to be an empty string due to the `getOrElse("")`.

Could we try pattern matching on `communicationAddress` here instead?